contracts/lens/Quoter.sol

```diff
@@ -0,0 +1,396 @@
+                poolKey: params.poolKey,
+                zeroForOne: params.zeroForOnes[i],
+                recipient: params.recipients[i],
+                amountOut: params.amountOuts[i],
+                sqrtPriceLimitX96: params.sqrtPriceLimitX96s[i],
+                hookData: params.hookData[i]
+            });
+            (int128[] memory deltaAmounts, uint160 sqrtPriceX96After, uint32 initializedTicksLoaded) =
+                quoteExactOutputSingle(singleParams);
+
+            deltas[i] = IQuoter.PoolDeltas({currency0Delta: deltaAmounts[0], currency1Delta: deltaAmounts[1]});
+            sqrtPriceX96AfterList[i] = sqrtPriceX96After;
+            initializedTicksLoadedList[i] = initializedTicksLoaded;
+        }
+    }
+
+    function lockAcquired(bytes calldata encodedSwapIntention) external returns (bytes memory) {
+        if (msg.sender != address(manager)) {
+            revert InvalidLockAcquiredSender();
+        }
+
+        SwapInfo memory swapInfo = abi.decode(encodedSwapIntention, (SwapInfo));
+        bytes memory result;
+
+        if (swapInfo.swapType == SwapType.ExactInputSingle) {
+            (BalanceDelta deltas, uint160 sqrtPriceX96After, int24 tickAfter) =
+                _quoteExactInputSingle(abi.decode(swapInfo.params, (ExactInputSingleParams)));
+
+            result = abi.encode(deltas, sqrtPriceX96After, tickAfter);
+        } else if (swapInfo.swapType == SwapType.ExactOutputSingle) {
+            (BalanceDelta deltas, uint160 sqrtPriceX96After, int24 tickAfter) =
+                _quoteExactOutputSingle(abi.decode(swapInfo.params, (ExactOutputSingleParams)));
+
+            result = abi.encode(deltas, sqrtPriceX96After, tickAfter);
+        } else if (swapInfo.swapType == SwapType.ExactInput) {
+            (
+                int128[] memory deltaAmounts,
+                uint160[] memory sqrtPriceX96AfterList,
+                uint32[] memory initializedTicksLoadedList
+            ) = _quoteExactInput(abi.decode(swapInfo.params, (ExactInputParams)));
+
+            result = abi.encode(deltaAmounts, sqrtPriceX96AfterList, initializedTicksLoadedList);
+        } else if (swapInfo.swapType == SwapType.ExactOutput) {
+            (
+                int128[] memory deltaAmounts,
+                uint160[] memory sqrtPriceX96AfterList,
+                uint32[] memory initializedTicksLoadedList
+            ) = _quoteExactOutput(abi.decode(swapInfo.params, (ExactOutputParams)));
+
+            result = abi.encode(deltaAmounts, sqrtPriceX96AfterList, initializedTicksLoadedList);
+        } else {
+            revert InvalidQuoteType();
+        }
+        assembly {
+            revert(add(0x20, result), mload(result))
+        }
+    }
+
+    function validateRevertReason(bytes memory reason) private pure returns (bytes memory) {
+        if (reason.length < MINIMUM_VALID_REASON_LENGTH) {
+            if (reason.length < MINIMUM_REASON_LENGTH) {
+                revert UnexpectedRevertBytes();
```

Comment:
I'd double check that this works if we are reverting with custom errors which are length 4

Reply:
good callout, added another branch checking that codition